readthedocs/config/config.py

```diff
@@ -825,19 +831,19 @@ def validate_final_doc_type(self):
         """
         dashboard_doctype = self.defaults.get('doctype', 'sphinx')
         if self.doctype != dashboard_doctype:
-            key = 'mkdocs' if self.doctype == 'mkdocs' else 'sphinx'
-            needed_doctype = (
-                'mkdocs' if dashboard_doctype == 'mkdocs' else 'sphinx'
-            )
-            self.error(
-                key,
-                'Your project is configured as "{doctype}" in your admin '
-                'dashboard, but there is no "{key}" key specified.'.format(
-                    doctype=dashboard_doctype,
-                    key=needed_doctype,
-                ),
-                code=INVALID_KEYS_COMBINATION,
-            )
+            error_msg = (
+                'Your project is configured as "{}" in your admin dashboard.'
+            ).format(self.builders_display[dashboard_doctype])
+
+            if dashboard_doctype == 'mkdocs' or not self.sphinx:
+                error_msg += ' But there is no "{}" key specified.'.format(
+                    'mkdocs' if dashboard_doctype == 'mkdocs' else 'sphinx'
+                )
+            else:
+                error_msg += ' But your "sphinx.builder" key does not match.'
```

Comment:
We're no longer catching the case where the user has no sphinx key at all. I still think this needs to be a separate check on top of checking if there is a `sphinx` key.

Reply:
The check for no sphinx key is above